method-README-template.md

```diff
@@ -1,65 +1,92 @@
+- *[data/revdictionary.json](https://github.com/taimoorkhan-nlp/latent_dirichlet_allocation/blob/master/data/revdictionary.json)*
+
+*scripts*
+- *[LDA-collapsed-gibbs-sampling.ipynb](https://github.com/taimoorkhan-nlp/latent_dirichlet_allocation/blob/master/LDA-collapsed-gibbs-sampling.ipynb)*
+- *[LICENSE.txt](https://github.com/taimoorkhan-nlp/latent_dirichlet_allocation/blob/master/LICENSE.txt)*
+- *requirements.txt not needed*
 
 ## Environment Setup
-- Setting up the envornment to run this method locally
+- Setting up the envornment to run the method locally
+- Deploy working environement from configuration files preserving dependencies i.e., Requirements.txt, install.R, configuration.yml etc.
 - Installing all the packages and libraries with specific versions required to run this method
-For example, 
-Requires Python 3.10.2 or 3.11.4. Suitable for environments focused on data collection and analysis.
-Dependencies are listed in requirements.txt and can be installed via pip install -r requirements.txt to ensure the tool functions correctly.
+- Load default configurations to execute the method without needing user envolvement e.g., for input or paths to resources. Store the configuration files separately (e.g., in config.json) where the users can change the paramters to alter the behavior of the method without directly interacting with the code.
 
+**Example:** *It is the vanilla implementation of the Latent Dirichlet Allocation technique with everything built from scratch, therefore only basic libraries i.e., `numpy`, `pandas`, `random` and `string` are needed to read data and generate random numbers.*
+- *Update [config.json](config.json) to read method configurations in JSON format and update as desired.* 
+- *Setup the environment using [requirements.txt](https://github.com/taimoorkhan-nlp/latent_dirichlet_allocation/blob/master/requirements.txt) through command `pip install -r requirements.txt`*
+- *Put your data in [data/input.csv](data/input.csv)*
+- *Execute the notebook [LDA-collapsed-gibbs-sampling.ipynb](LDA-collapsed-gibbs-sampling.ipynb) to get results*
 
-## Hardware Requirements (Optional)
-- The hardware requirements may be needed in specific cases when a method is known to require more memory/compute power. 
-- The method need to be executed on a specific architecture (GPUs, Hadoop cluster etc.)
+## Input and Output Data
+- Explain the input data and provide links to the dataset that the method can work with. It should idealy be [GESIS DBD data](https://www.gesis.org/en/institute/digital-behavioral-data)
+- Explain the sample input data in the repo i.e., the data features and what they represent
+- Explain the sample output data in the repo i.e., how to interpret the output  
 
+## How to Use
+- Provide list of steps to execute the method for its intended purpose (assuming the environment is deployed)
+- Explain which files or paramters need to be update dand how the alter the behavior of the method
```

Comment:
Typo here.

- [ ] "update dand" should be "updated and".
- [ ] "paramters" should be "parameters".